### PR TITLE
Housekeeping Update to Net 9.0

### PR DESCRIPTION
**Pull request by ChrisPulman**
Update to use Net Standard 2.0, Net framework 4.6.2, Net 8.0, and Net 9.0

### CODE REVIEW

File: src/DynamicData.Tests/Cache/EditDiffChangeSetOptionalFixture.cs [hunk 1]
```diff
@@ -190,7 +190,7 @@ private class PersonComparer : IEqualityComparer<Person>
     {
         public bool Equals([DisallowNull] Person x, [DisallowNull] Person y) =>
             EqualityComparer<string>.Default.Equals(x.Name, y.Name) && EqualityComparer<int>.Default.Equals(x.Id, y.Id);
-        public int GetHashCode([DisallowNull] Person obj) => throw new NotImplementedException();
+        public int GetHashCode([DisallowNull] Person obj) => obj.GetHashCode();
```
Comment (JakenVeina) [inline src/DynamicData.Tests/Cache/EditDiffChangeSetOptionalFixture.cs:193]: I'm guessing this (and the others) started failing some test with .NET 9?

**Review by JakenVeina — COMMENTED** [review]
In general, I'm really not a fan of having so many subjective, cosmetic changes mixed in here, that have nothing to do with a .NET 9 upgrade. Especially since, for many of them, the choice to not use more-concise syntax was intentional.

File: src/DynamicData.Tests/API/ApiApprovalTests.DynamicDataTests.DotNet8_0.verified.txt [hunk 2]
```diff
@@ -469,7 +469,7 @@ namespace DynamicData.Binding
         public static bool operator !=(DynamicData.Binding.PropertyValue<TObject, TValue>? left, DynamicData.Binding.PropertyValue<TObject, TValue>? right) { }
         public static bool operator ==(DynamicData.Binding.PropertyValue<TObject, TValue>? left, DynamicData.Binding.PropertyValue<TObject, TValue>? right) { }
     }
-    public struct SortAndBindOptions : System.IEquatable<DynamicData.Binding.SortAndBindOptions>
+    public readonly struct SortAndBindOptions : System.IEquatable<DynamicData.Binding.SortAndBindOptions>
```
[the remark this answers is not in the document]
Comment (ChrisPulman) [reply src/DynamicData.Tests/API/ApiApprovalTests.DynamicDataTests.DotNet8_0.verified.txt:472]: Not an error, just a performance recommendation for this one, along with a few others

File: src/DynamicData.Tests/Cache/EditDiffChangeSetOptionalFixture.cs (hunk 1 above)
Comment (ChrisPulman) [reply src/DynamicData.Tests/Cache/EditDiffChangeSetOptionalFixture.cs:193]: Yes, I didn't see any test's specifically looking for the not implemented exception for the GetHashcode ones, so figured implementing was the better option

**Pull request closed without merging**